docs/en_US/TrainingService/HeterogeneousMode.md

```diff
@@ -0,0 +1,54 @@
+**Run an Experiment on Heterogeneous Mode**
+===
+Run NNI on heterogeneous mode means that NNI will run trials jobs in multiple kinds of training platforms. For example, NNI could submit trial jobs to remote machine and AML simultaneously。
+
+## Setup environment
+NNI has supported [local](./LocalMode.md), [remote](./RemoteMachineMode.md), [pai](./PaiMode.md) and [AML](./AMLMode.md) for heterogeneous training service. Before starting an experiment using these mode, users should setup the corresponding environment for the platforms. More details about the environment setup could be found in the corresponding docs.
+
+
+
+## Run an experiment
+Use `examples/trials/mnist-tfv1` as an example. The NNI config YAML file's content is like:
+
+```yaml
+authorName: default
+experimentName: example_mnist
+trialConcurrency: 2
+maxExecDuration: 1h
+maxTrialNum: 10
+trainingServicePlatform: heterogeneous
+searchSpacePath: search_space.json
+#choice: true, false
+useAnnotation: false
+tuner:
+  #choice: TPE, Random, Anneal, Evolution, BatchTuner, MetisTuner, GPTuner
+  #SMAC (SMAC should be installed through nnictl)
+  builtinTunerName: TPE
+  classArgs:
+    #choice: maximize, minimize
+    optimize_mode: maximize
+trial:
+  command: python3 mnist.py
+  codeDir: .
+  gpuNum: 1
+heterogeneousConfig:
+  trainingServicePlatforms:
```

Comment:
Propose to write as below format. So that user can input multiple AML or PAI instances. And the `trainingServicePlatforms` is not necessary also.

```yaml
heterogeneousConfig
    - type: local
    - type: remote
      machineList:
      - ip: 10.1.1.1
        username: bob
        passwd: bob123
        #port can be skip if using default ssh port 22
        #port: 22
```

Reply:
As I know, `heterogeneousConfig` will not be exposed to users, users will use 
```
trainingServiceConfig:
    - platform: local
    - platform: remote
      machineList:
```
for heterogeneous purpose. This pr will keep current heterogenerouConfig format, and we will have another pr to convert new yaml file(exposed to users) to current yaml file(internal use). Add @liuzhe-lz for confirm.